cpp/src/arrow/flight/types.cc

```diff
@@ -151,18 +151,57 @@ Status MakeFlightError(FlightStatusCode code, std::string message,
                        std::make_shared<FlightStatusDetail>(code, std::move(extra_info)));
 }
 
-bool FlightDescriptor::Equals(const FlightDescriptor& other) const {
-  if (type != other.type) {
-    return false;
+static std::ostream& operator<<(std::ostream& os, std::vector<std::string> values) {
```

Comment:
The `vector` here needs to be taken by `const &`. I will do this in another PR. I just moved this function.